Weather/Views/WeatherViewController.swift

```diff
@@ -44,7 +40,10 @@ extension WeatherViewController: UICollectionViewDataSource, UICollectionViewDel
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
         let view = self.view as! WeatherView
```

Comment:
Avoid force casts as well.
In this case you can use pattern matching with switch statement. Use type casting in switch cases.